lib/mix/lib/mix/tasks/local.rebar.ex

```diff
@@ -25,13 +27,28 @@ defmodule Mix.Tasks.Local.Rebar do
       []       -> @rebar_url
       [path|_] -> path
     end
+
     do_install(path, opts)
   end
 
   defp do_install(path, opts) do
     local_rebar_path = Mix.Rebar.local_rebar_path
-
-    if Mix.Utils.copy_path!(path, local_rebar_path, opts) do
+    tmp_local_rebar_path = local_rebar_path <> ".tmp"
+
+    File.rm(tmp_local_rebar_path)
+
+    if Mix.Utils.copy_path!(path, tmp_local_rebar_path, opts) do
```

Comment:
Since we need to load into memory anyway, we should use `Mix.Utils.read_path!` and write it down once when we confirm it is valid.